ksql-api/src/main/java/io/confluent/ksql/api/server/Server.java

```diff
@@ -93,22 +93,20 @@ public synchronized void stop() {
     }
   }
 
-  ApiQueryID registerQuery(final QuerySubscriber querySubscriber) {
-    Objects.requireNonNull(querySubscriber);
-    final ApiQueryID queryID = new ApiQueryID();
-    if (queries.putIfAbsent(queryID, querySubscriber) != null) {
+  void registerQuery(final PushQueryHolder query) {
+    Objects.requireNonNull(query);
+    if (queries.putIfAbsent(query.getId(), query) != null) {
       // It should never happen
       // https://stackoverflow.com/questions/2513573/how-good-is-javas-uuid-randomuuid
       throw new IllegalStateException("Glitch in the matrix");
     }
-    return queryID;
   }
 
-  QuerySubscriber removeQuery(final ApiQueryID queryID) {
-    return queries.remove(queryID);
+  PushQueryHolder removeQuery(final PushQueryId queryId) {
+    return queries.remove(queryId);
```

Comment:
with reference to this comment: https://github.com/confluentinc/ksql/pull/4376#discussion_r370589845

We really do try and avoid returning `nulls` in the code by encoding optionality into the type system.  Please use `Optional`. Pretty please?